src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -2309,10 +2322,15 @@ private bool EmitAssignmentPreamble(BoundAssignmentOperator assignmentOperator)
                 case BoundKind.FieldAccess:
                     {
                         var left = (BoundFieldAccess)assignmentTarget;
-                        if (!left.FieldSymbol.IsStatic)
+                        if (left.FieldSymbol.RefKind != RefKind.None &&
+                            !assignmentOperator.IsRef)
+                        {
+                            EmitFieldLoadNoIndirection(left, used: true);
+                        }
+                        else if (!left.FieldSymbol.IsStatic)
                         {
                             var temp = EmitReceiverRef(left.ReceiverOpt, AddressKind.Writeable);
-                            Debug.Assert(temp == null, "temp is unexpected when assigning to a field");
```

Comment:
>Debug.Assert(temp == null, "temp is unexpected when assigning to a field"); [](http://example.com/codeflow?start=28&length=75)

Is there a specific reason to drop the assert?

Reply:
The receiver `new S<int>()` in the following results in a temporary:
```csharp
int i = 42;
new S<int>().F = ref i;

ref struct S<T>
{
    public ref T F;
}
```